sklearn/ensemble/_weight_boosting.py

```diff
@@ -492,11 +502,16 @@ class AdaBoostClassifier(ClassifierMixin, BaseWeightBoosting):
     0.983...
     """
 
+    # TODO(1.6): Remove _parameter_constraints for "algorithm"
     _parameter_constraints: dict = {
         **BaseWeightBoosting._parameter_constraints,
-        "algorithm": [StrOptions({"SAMME", "SAMME.R"})],
+        "algorithm": [
+            StrOptions({"SAMME", "SAMME.R"}),
+            Hidden(StrOptions({"deprecated"})),
```

Comment:
Actually, we don't need the `"deprecated"` option in this deprecation. We always want to warn if the default is as-is or if the user pass `"SAMME.R"`. We can remove this line. We would use `"deprecated"` if we would like to entirely remove the `algorithm` parameter here.

Reply:
I was wondering about this. Thanks for clearing this up.